execution/state.go

```diff
@@ -54,15 +54,18 @@ var (
 	storageKeyFormat  = storage.NewMustKeyFormat("s", crypto.AddressLength, binary.Word256Length)
 	nameKeyFormat     = storage.NewMustKeyFormat("n", storage.VariadicSegmentLength)
 	proposalKeyFormat = storage.NewMustKeyFormat("p", sha256.Size)
+
 	// Keys that reference references
 	blockRefKeyFormat = storage.NewMustKeyFormat("b", uint64Length)
 	txRefKeyFormat    = storage.NewMustKeyFormat("t", uint64Length, uint64Length)
-	// Reference keys
+
+	// Reference keys (that do not contribute to state hash)
 	// TODO: implement content-addressing of code and optionally blocks (to allow reference to block to be stored in state tree)
 	//codeKeyFormat   = storage.NewMustKeyFormat("c", sha256.Size)
 	//blockKeyFormat  = storage.NewMustKeyFormat("b", sha256.Size)
-	txKeyFormat     = storage.NewMustKeyFormat("b", tmhash.Size)
-	commitKeyFormat = storage.NewMustKeyFormat("x", tmhash.Size)
+	txKeyFormat = storage.NewMustKeyFormat("b", tmhash.Size)
+	// Binding between apphash and version stto
+	commitKeyFormat = storage.NewMustKeyFormat("v", uint64Length)
```

Comment:
commitKeyFormat changed, this is used during startup to load the correct apphash. How can this be compatible?

Reply:
yep... it won't be, this will need a fresh sync, would have been worth mentioning